UnitsNet/CustomCode/UnitParser.cs

```diff
@@ -94,6 +100,34 @@ public Enum Parse([NotNull] string unitAbbreviation, Type unitType, [CanBeNull]
             }
         }
 
+        internal static string NormalizeUnitString(string unitAbbreviation)
+        {
+            // remove all whitespace in the string
+            unitAbbreviation = new string(unitAbbreviation.Where(c => !char.IsWhiteSpace(c)).ToArray());
```

Comment:
I think you asked about feedback on this method in particular.

As far as I know, the way you strip whitespace here is perfectly fine. You could use regex, but I suspect that will just be slower and not really bring any new value.

The below `.Replace()` calls can be optimized in terms of string copying, but we are talking extremely small strings here - typically less than 10 chars. I'm sure there are clever ways to reduce string allocations instead of many calls to `string.Replace()`, but I don't know of any and not sure it's worth much effort in this case.

If you just want the code to be shorter, you can use "method chaining" like this. This does not bring any optimizations though, only shorter syntax.
```c#
unitAbbreviation = unitAbbreviation
    .Replace("^-9", "⁻⁹")
    .Replace("^-8", "⁻⁸")
    .Replace("^-7", "⁻⁷");
```